apps/balancer-tools/src/app/apr/round/(components)/PoolListTable.tsx

```diff
@@ -9,69 +9,63 @@ import {
 import { Tooltip } from "#/components/Tooltip";
-import { Pool } from "#/lib/balancer/gauges";
 import { fetcher } from "#/utils/fetcher";
 import { formatNumber } from "#/utils/formatNumber";
 
 import { PoolTypeEnum } from "../../(utils)/calculatePoolStats";
-import { PoolStatsData, PoolStatsResults } from "../../api/route";
+import { PoolStatsData, PoolStatsResults, PoolTokens } from "../../api/route";
+import { TokenFilterInput } from "./TokenFilterInput";
 
 export function PoolListTable({
   roundId,
   initialData,
 }: {
   roundId: string;
-  initialData: PoolStatsResults;
+  initialData: PoolStatsData[];
 }) {
-  const [tableData, setTableData] = useState(initialData.perRound);
+  const router = useRouter();
+  const searchParams = useSearchParams();
+  const [tableData, setTableData] = useState(initialData);
   const [isLoadingMore, setIsLoadingMore] = useState(false);
-  const [sortField, setSortField] = useState("apr");
-  const [order, setOrder] = useState<"asc" | "desc">("desc");
-
-  const handleSorting = (sortField: keyof PoolStatsData, sortOrder: string) => {
-    if (sortField) {
-      setTableData((prevTableData) => {
-        const sortedArray = prevTableData.slice().sort((a, b) => {
-          const aValue = a[sortField] as number | string;
-          const bValue = b[sortField] as number | string;
-
-          // Handle NaN values
-          if (typeof aValue === "number" && isNaN(aValue)) return 1;
-          if (typeof bValue === "number" && isNaN(bValue)) return -1;
-
-          if (aValue < bValue) {
-            return sortOrder === "asc" ? -1 : 1;
-          } else if (aValue > bValue) {
-            return sortOrder === "asc" ? 1 : -1;
-          }
-          return 0;
-        });
-
-        return sortedArray;
-      });
-    }
-  };
+
+  useEffect(() => {
+    setTableData(initialData);
+  }, [initialData]);
 
   const handleSortingChange = (accessor: keyof PoolStatsData) => {
     const sortOrder =
-      accessor === sortField && order === "desc" ? "asc" : "desc";
-    setSortField(accessor);
-    setOrder(sortOrder);
-    handleSorting(accessor, sortOrder);
+      accessor === searchParams.get("sort") &&
+      searchParams.get("order") === "desc"
+        ? "asc"
+        : "desc";
+    const current = new URLSearchParams(Array.from(searchParams.entries()));
+    current.set("sort", accessor);
+    current.set("order", sortOrder);
+
+    const search = current.toString();
+    const query = search ? `?${search}` : "";
+    router.push(window.location.pathname + query, { scroll: false });
   };
 
   const loadMorePools = async () => {
     setIsLoadingMore(true);
     const aditionalPoolsData = await fetcher<PoolStatsResults>(
       `${
         process.env.NEXT_PUBLIC_SITE_URL
-      }/apr/api/?roundId=${roundId}&sort=${sortField}&order=${order}&limit=10&offset=${
+      }/apr/api/?roundId=${roundId}&sort=${searchParams.get(
+        "sort",
+      )}&order=${searchParams.get("order")}&limit=10&offset=${
```

Comment:
`#thought` here we should just forward whatever is in the URL. And whenever we show a page we should probably load it with these parameters. We could add a middleware to do that. wdyt?

Reply:
Yeah, this is going to break with the addition of more filters, but I imagine that in the should look something like this ``${ process.env.NEXT_PUBLIC_SITE_URL }/apr/api/?${...searchParams}&offset=${Object.keys(tableData).length}``